### DISCUSSION TITLE
v3.5.0-M1

### DISCUSSION THREAD

**Discussion by nafg** [slick/slick#2635] [Announcements]
Note: Java 8 support was restored after it had been initially dropped during this cycle.

### Highlights 🎉
* Drop support for old SQL Server versions by @nafg in https://github.com/slick/slick/pull/2558
* Insert multiple rows using a single SQL statement by @jkugiya in https://github.com/slick/slick/pull/2398
* Adds insertOrUpdateAll by @jkugiya in https://github.com/slick/slick/pull/2397
* Fix SQL Server tests. Drops JTDS by @nafg in https://github.com/slick/slick/pull/2560
### Enhancements ✨
* Attach base statement as child for asTry dumpInfo by @natefitzgerald in https://github.com/slick/slick/pull/2584
* More type-safe way of expressing profiles' support of RowsPerStatement.All by @nafg in https://github.com/slick/slick/pull/2612
* Add exceptionOverrideClassName config for HikariCP by @jsoizo in https://github.com/slick/slick/pull/2621
### Bugs fixed 🐛️
* Handle in-use count errors by @nafg in https://github.com/slick/slick/pull/2619
* BasicBackend#DatabaseDef#releaseSession: null out the session first, … by @nafg in https://github.com/slick/slick/pull/2620
### Documentation improvements 📃️
* Document using the generic jdbc profile by @ElectricWound in https://github.com/slick/slick/pull/2586
* build.sbt: set site description setting by @nafg in https://github.com/slick/slick/pull/2623
### Chores 🛠
* Update slf4j-api to 2.0.0 by @scala-steward in https://github.com/slick/slick/pull/2549
* Update slf4j-api to 2.0.1 by @scala-steward in https://github.com/slick/slick/pull/2554
* Update dependency com.typesafe:sbt-mima-plugin to v1.1.1 by @renovate in https://github.com/slick/slick/pull/2555
* Drop Java 8 support by @nafg in https://github.com/slick/slick/pull/2557
* CompatReportPlugin: use details/summary tags by @nafg in https://github.com/slick/slick/pull/2559
* Update sbt-scoverage to 2.0.3 by @scala-steward in https://github.com/slick/slick/pull/2553
* Update logback-classic to 1.4.1 by @scala-steward in https://github.com/slick/slick/pull/2552
* Update dependency com.zaxxer:HikariCP to v5 by @renovate in https://github.com/slick/slick/pull/2463
* Update scala-compiler, scala-library to 2.12.17 by @scala-steward in https://github.com/slick/slick/pull/2556
* Update dependency org.scoverage:sbt-scoverage to v2.0.4 by @renovate in https://github.com/slick/slick/pull/2561
* Update dependency org.slf4j:slf4j-api to v2.0.2 by @renovate in https://github.com/slick/slick/pull/2562
* Update dependency org.slf4j:slf4j-api to v2.0.3 by @renovate in https://github.com/slick/slick/pull/2563
* build.sbt: versionPolicyIntention: BumpMajor by @nafg in https://github.com/slick/slick/pull/2565
* build.sbt: testAll should run versionPolicyCheck on actual modules by @nafg in https://github.com/slick/slick/pull/2566
* Update dependency org.scalatestplus:testng-7-5 to v3.2.14.0 by @renovate in https://github.com/slick/slick/pull/2564
* Update dependency ch.qos.logback:logback-classic to v1.4.2 by @renovate in https://github.com/slick/slick/pull/2568
* Update dependency ch.qos.logback:logback-classic to v1.4.3 by @renovate in https://github.com/slick/slick/pull/2569
* Update dependency ch.qos.logback:logback-classic to v1.4.4 by @renovate in https://github.com/slick/slick/pull/2574
* Update dependency org.scoverage:sbt-scoverage to v2.0.5 by @renovate in https://github.com/slick/slick/pull/2576
* Update hsqldb to 2.7.0 by @scala-steward in https://github.com/slick/slick/pull/2511
* Update sbt to 1.7.2 by @scala-steward in https://github.com/slick/slick/pull/2582
* Update ibmcom/db2 Docker tag to v11.5.8.0 by @renovate in https://github.com/slick/slick/pull/2585
* Update mysql-connector-java to 8.0.31 by @scala-steward in https://github.com/slick/slick/pull/2580
* Update dependency org.scoverage:sbt-scoverage to v2.0.6 by @renovate in https://github.com/slick/slick/pull/2587
* .mergify.yml: don't require signing Lightbend CLA by @nafg in https://github.com/slick/slick/pull/2590
* Update dependency org.hsqldb:hsqldb to v2.7.1 by @renovate in https://github.com/slick/slick/pull/2592
* Use scala version wildcards for ci.yml matrix by @nafg in https://github.com/slick/slick/pull/2594
* Update sbt-ci-release to 1.5.11 by @scala-steward in https://github.com/slick/slick/pull/2593
* Update scala-compiler, scala-library to 2.13.10 by @slick-scala-steward in https://github.com/slick/slick/pull/2577
* Update paradox-theme-generic, sbt-paradox to 0.10.3 by @slick-scala-steward in https://github.com/slick/slick/pull/2596
* Update dependency ch.epfl.scala:sbt-version-policy to v2.1.0 by @renovate in https://github.com/slick/slick/pull/2597
* use `CollectionConverters` instead of deprecated `JavaConverters` by @xuwei-k in https://github.com/slick/slick/pull/2599
* add explicit types for implicits by @xuwei-k in https://github.com/slick/slick/pull/2598
* Update sbt to 1.7.3 by @slick-scala-steward in https://github.com/slick/slick/pull/2600
* Revert "Update sbt to 1.7.3" by @nafg in https://github.com/slick/slick/pull/2601
* Update sqlite-jdbc to 3.39.4.0 by @slick-scala-steward in https://github.com/slick/slick/pull/2603
* Update sbt to 1.7.3 by @scala-steward in https://github.com/slick/slick/pull/2604
* Update sbt to 1.8.0 by @slick-scala-steward in https://github.com/slick/slick/pull/2605
* Update dependency org.xerial:sqlite-jdbc to v3.39.4.1 by @renovate in https://github.com/slick/slick/pull/2606
* Update dependency org.slf4j:slf4j-api to v2.0.4 by @renovate in https://github.com/slick/slick/pull/2607
* Update dependency ch.qos.logback:logback-classic to v1.4.5 by @renovate in https://github.com/slick/slick/pull/2608
* Update dependency org.xerial:sqlite-jdbc to v3.40.0.0 by @renovate in https://github.com/slick/slick/pull/2609
* Update dependency org.postgresql:postgresql to v42.5.1 by @renovate in https://github.com/slick/slick/pull/2610
* Update dependency org.slf4j:slf4j-api to v2.0.5 by @renovate in https://github.com/slick/slick/pull/2611
* Update scala-collection-compat to 2.9.0 by @slick-scala-steward in https://github.com/slick/slick/pull/2613
* pr-compat-report-generate.yml: use -no-colors and switch to GH Environment Files by @nafg in https://github.com/slick/slick/pull/2616
* pr-compat-report-generate.yml: give up on piping and do the writing within SBT by @nafg in https://github.com/slick/slick/pull/2617
* Update dependency com.oracle.database.jdbc.debug:ojdbc8_g to v21.8.0.0 by @renovate in https://github.com/slick/slick/pull/2618
* Update dependency org.slf4j:slf4j-api to v2.0.6 by @renovate in https://github.com/slick/slick/pull/2624
* Restore Java 8 support by @nafg in https://github.com/slick/slick/pull/2622
* Update dependency org.scalatestplus:testng-7-5 to v3.2.15.0 by @renovate in https://github.com/slick/slick/pull/2629
* Update sbt to 1.8.2 by @scala-steward in https://github.com/slick/slick/pull/2630
* Publish with JDK 8 again by @nafg in https://github.com/slick/slick/pull/2632

## New Contributors
* @ElectricWound made their first contribution in https://github.com/slick/slick/pull/2586
* @natefitzgerald made their first contribution in https://github.com/slick/slick/pull/2584
* @slick-scala-steward made their first contribution in https://github.com/slick/slick/pull/2577
* @jsoizo made their first contribution in https://github.com/slick/slick/pull/2621

**Full Changelog**: https://github.com/slick/slick/compare/v3.4.1...v3.5.0-M1

<hr /><em>This discussion was created from the release <a href='https://github.com/slick/slick/releases/tag/v3.5.0-M1'>v3.5.0-M1</a>.</em>

**Comment by luksow** [comment]
Hi, first of all, thanks for keeping Slick alive, much appreciated!

My question here is at which point 3.5.0-Mx will get a Scala 3 build?

**Reply by nafg** [reply]
That's my current focus (when I can spend time on Slick).
